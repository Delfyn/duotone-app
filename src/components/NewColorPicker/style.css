.new-color-picker {
  border-top: 1px solid var(--border-color);
}
