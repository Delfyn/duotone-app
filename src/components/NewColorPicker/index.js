/* @flow */

import React from 'react';
import classNames from 'classnames';
import ColorPicker from '../ColorPicker';
import FlipArrow from '../FlipArrow';
import Checkmark from '../Checkmark';
import './style.css';

type Props = {
  className?: string,
  primary: string,
  secondary: string,
  onAddDuotoneClick: Function,
  onSetDuotoneClick: Function,
  onSetPrimaryClick: Function,
  onSetSecondaryClick: Function,
};

const NewColorPicker = ({
  className,
  primary,
  secondary,
  onAddDuotoneClick,
  onSetDuotoneClick,
  onSetPrimaryClick,
  onSetSecondaryClick,
}: Props) => (
  <div className={classNames('new-color-picker flex pt2', className)}>
    <ColorPicker className="mr1" color={primary} onChange={onSetPrimaryClick} />
    <ColorPicker className="mr1" color={secondary} onChange={onSetSecondaryClick} />
    <FlipArrow
      className="mr1"
      onClick={() => onSetDuotoneClick(secondary, primary)}
    />
    <Checkmark
      onClick={() => onAddDuotoneClick(primary, secondary)}
    />
  </div>
);

export default NewColorPicker;
