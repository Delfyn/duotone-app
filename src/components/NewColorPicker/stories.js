/* @flow */

import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import NewColorPicker from './index';

storiesOf('NewColorPicker', module)
  .add('Simple example', () => (
    <NewColorPicker
      primary="#fd3659"
      secondary="#243261"
      onAddDuotoneClick={action('onAddDuotoneClick')}
      onSetDuotoneClick={action('onSetDuotoneClick')}
      onSetPrimaryClick={action('onSetPrimaryClick')}
      onSetSecondaryClick={action('onSetSecondaryClick')}
    />
  ));
