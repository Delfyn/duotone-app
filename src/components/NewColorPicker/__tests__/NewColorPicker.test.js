import React from 'react';
import renderer from 'react-test-renderer';
import NewColorPicker from '../index';

test('NewColorPicker renders correctly', () => {
  const component = renderer.create(
    <NewColorPicker />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
