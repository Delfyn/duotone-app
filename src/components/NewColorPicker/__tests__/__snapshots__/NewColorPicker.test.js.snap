exports[`test NewColorPicker renders correctly 1`] = `
<div
  className="new-color-picker flex pt2">
  <div>
    <div
      className="color-field flex-auto mr1"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
        }
      } />
  </div>
  <div>
    <div
      className="color-field flex-auto mr1"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
        }
      } />
  </div>
  <svg
    className="mr1"
    height={40}
    onClick={[Function]}
    viewBox="0 0 40 40"
    width={40}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink">
    <defs>
      <circle
        cx={20}
        cy={20}
        id="a"
        r={20} />
      <mask
        fill="#fff"
        height={40}
        id="b"
        width={40}
        x="0"
        y="0">
        <use
          xlinkHref="#a" />
      </mask>
    </defs>
    <g
      fill="none"
      fillRule="evenodd">
      <use
        mask="url(#b)"
        stroke="#979797"
        strokeWidth="4"
        xlinkHref="#a" />
      <g
        fill="#9B9B9B">
        <path
          d="M8 20l5-7v14zM13 21v-2h14v2zM32 20l-5 7V13z" />
      </g>
    </g>
  </svg>
  <svg
    className=""
    height={40}
    onClick={[Function]}
    viewBox="0 0 40 40"
    width={40}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink">
    <defs>
      <circle
        cx={20}
        cy={20}
        id="a"
        r={20} />
      <mask
        fill="#fff"
        height={40}
        id="b"
        width={40}
        x="0"
        y="0">
        <use
          xlinkHref="#a" />
      </mask>
    </defs>
    <g
      fill="#333"
      fillRule="evenodd">
      <use
        mask="url(#b)"
        stroke="#333"
        strokeWidth="4"
        xlinkHref="#a" />
      <path
        d="M17.854 26.96l12.57-12.57-1.414-1.415-11.156 11.157-6.157-6.157-1.415 1.414z"
        fill="#fff" />
    </g>
  </svg>
</div>
`;
